web/src/components/netflow-record/record-panel.tsx

```diff
@@ -0,0 +1,142 @@
+  setRange: (r: number | TimeRange) => void;
+  setQueryOptions: (opts: QueryOptions) => void;
+  onClose: () => void;
+  id?: string;
+};
+
+export const RecordPanel: React.FC<RecordDrawerProps> = ({
+  id,
+  record,
+  columns,
+  filters,
+  range,
+  options,
+  setFilters,
+  setRange,
+  setQueryOptions,
+  onClose
+}) => {
+  const { t } = useTranslation('plugin__network-observability-plugin');
+
+  const onClick = React.useCallback(
+    (col: Column, isDelete: boolean) => {
+      if (!record) {
+        return;
+      }
+
+      if (isDelete) {
+        switch (col.id) {
+          case ColumnsId.timestamp:
+            setRange(DEFAULT_TIME_RANGE);
+            break;
+          case ColumnsId.flowdir:
+            setQueryOptions({ ...options, reporter: flowdirToReporter[DEFAULT_FLOWDIR] });
+            break;
+          default:
+            setFilters(filters.filter((f: Filter) => f.colId !== col.id));
+            break;
+        }
+      } else {
+        const value = col.value(record);
+        switch (col.id) {
+          case ColumnsId.timestamp:
+            //Filter at exact same date in ms
+            const dateSeconds = getDateMsInSeconds(Number(value));
+            setRange({ from: dateSeconds, to: dateSeconds + 1 });
+            break;
+          case ColumnsId.flowdir:
+            setQueryOptions({ ...options, reporter: flowdirToReporter[value] });
+            break;
+          default:
+            const values = [{ v: value.toString() }];
+            const result = _.cloneDeep(filters);
+            const found = result.find(f => f.colId === col.id);
+            if (found) {
+              found.values = values;
+            } else {
+              result.push({ colId: col.id, values: values });
+            }
+            setFilters(result);
+            break;
+        }
+      }
+    },
+    [filters, options, record, setFilters, setQueryOptions, setRange]
+  );
+
+  const getFilter = (col: Column) => {
+    let isDelete = false;
+    if (record) {
+      const value = col.value(record);
+      switch (col.id) {
+        case ColumnsId.timestamp:
+          isDelete = typeof range !== 'number' && range.from === getDateMsInSeconds(Number(value));
+          break;
+        case ColumnsId.flowdir:
+          isDelete = options.reporter === flowdirToReporter[value];
+          break;
+        default:
+          isDelete =
+            filters.find((f: Filter) => f.colId === col.id && f.values.find(v => v.v === value.toString())) !==
+            undefined;
+          break;
+      }
+    }
+    return {
+      onClick,
+      isDelete
+    };
+  };
+
+  return (
+    <DrawerPanelContent id={id}>
+      <DrawerHead>
+        <Text component={TextVariants.h2}>{t('Traffic details')}</Text>
+        <DrawerActions>
+          <DrawerCloseButton onClick={onClose} />
+        </DrawerActions>
+      </DrawerHead>
+      <DrawerPanelBody>
+        {record &&
+          columns.map(c => (
```

Comment:
The intent here is to show all fields in `record`, not only the ones that map to a column. It allows to get the full flow regardless what we consider valuable for table display.
For instance, I should be able to look at the mac addresses even if the option is not available in columns list.

Reply:
(I was thinking at first about something very simple showing flow key / values, without even bothering about translating keys .. although translating known keys is still nice to have)

Reply:
I had doubt about that during development. How do you see the missing fields ? 
Should we :
- add missing record key / values at the end of the panel so all the missing fields will be visible but not having good labels
- add extra columns that can't be displayed in the table but appear in the panel; then some fields may be missing
- add a [raw json](https://www.patternfly.org/v4/components/clipboard-copy/#json-object-wrap-code-with-pre) at the end of the panel; less friendly but easy to maintain

Reply:
yeah maybe the raw json alternative is good enough, simpler and more reliable

Reply:
(more especially as if I'm correct, we can't do filtering with non-column fields.. well at least given the current filters implementation)